diff --git a/openvaf/llvm/src/initialization.rs b/openvaf/llvm/src/initialization.rs
index 7d641cb..8bf118e 100644
--- a/openvaf/llvm/src/initialization.rs
+++ b/openvaf/llvm/src/initialization.rs
@@ -102,7 +102,7 @@ unsafe fn configure_llvm(cg_opts: &[String], tg_opts: &[String]) {
     LLVMParseCommandLineOptions(
         llvm_args.len() as c_int,
         llvm_args.as_ptr(),
-        b"".as_ptr() as *const c_char,
+        b"\0".as_ptr() as *const c_char,  // MXL: was b"", a dangling zero-sized ptr -> strlen(0x1)
     );
 }
 
